File: service/nexus/src/main/java/io/xj/service/nexus/fabricator/FabricationException.java
// Copyright (c) XJ Music Inc. (https://xj.io) All Rights Reserved.

package io.xj.service.nexus.fabricator;

import io.xj.lib.util.Text;

public class FabricationException extends Exception {

  public FabricationException(String msg) {
    super(msg);
  }

  public FabricationException(String msg, Exception e) {
    super(String.format("%s %s\n%s", msg, e.toString(), Text.formatStackTrace(e)));
  }

  public FabricationException(Throwable targetException) {
    super(String.format("%s\n%s", targetException.getMessage(), Text.formatStackTrace(targetException)));
  }
}

File: engine/src/fabricator/FabricationException.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include "xjmusic/fabricator/FabricationException.h"

using namespace XJ;


FabricationException::FabricationException(std::string msg) : msg_(std::move(msg)) {}


const char *FabricationException::what() const noexcept {
  return msg_.c_str();
}


